### DISCUSSION TITLE
About Catppuccin

### DISCUSSION THREAD

**Discussion by fdelacruz** [LazyVim/LazyVim#2269] [Q&A] [answered]
In the [coloscheme](https://www.lazyvim.org/plugins/colorscheme) section of Lazyvim there's the following code:
```
{
  "catppuccin/nvim",
  lazy = true,
  name = "catppuccin",
  opts = {
    integrations = {
      aerial = true,
      alpha = true,
      cmp = true,
      dashboard = true,
      flash = true,
      gitsigns = true,
      headlines = true,
      illuminate = true,
      indent_blankline = { enabled = true },
      leap = true,
      lsp_trouble = true,
      mason = true,
      markdown = true,
      mini = true,
      native_lsp = {
        enabled = true,
        underlines = {
          errors = { "undercurl" },
          hints = { "undercurl" },
          warnings = { "undercurl" },
          information = { "undercurl" },
        },
      },
      navic = { enabled = true, custom_bg = "lualine" },
      neotest = true,
      neotree = true,
      noice = true,
      notify = true,
      semantic_tokens = true,
      telescope = true,
      treesitter = true,
      treesitter_context = true,
      which_key = true,
    },
  },
}
```
Which I added in my `plugins/catppuccin.lua` file
Now these are the same settings already here `LazyVim/lua/lazyvim/plugins/colorscheme.lua` so I figure that is only needed in the case I want to change individual options.
The issue that if I delete the file then catppuccin doesn't load its colors properly.

But, if is add
```
return {
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
    }
  }
}
```
then it is back to normal. Also i dont have two catppuccin AND nvim/catppucin lazy loading 
Would that be the proper to load the already installed by default catppuccin??

**Comment by dpetka2001** [comment]
The settings in the Lazyvim docs are the defaults and there is no need for you to copy them over. Only if you want to change something.

**Reply by fdelacruz** [reply]
that was from [icons & colorscheme](https://www.lazyvim.org/configuration#icons--colorscheme) `lua/plugins/core.lua`
```
return {
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
    }
  }
}
```
It’s that right way setup the default colorscheme?? It’s not clear how to choose between tokyonight and cappuccin in the setup docs.

**Reply by dpetka2001** [reply]
Yes, that's the right way to do it.
What do you mean it's not clear in the docs? There's an example about changing colorscheme to gruvbox. The same applies for all colorschemes.

**Reply by fdelacruz** [reply]
the [docs](https://www.lazyvim.org/plugins/colorscheme) state that one should use `lua/plugins/colorscheme.lua`:
```
return {
  -- add gruvbox
  { "ellisonleao/gruvbox.nvim" },

  -- Configure LazyVim to load gruvbox
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "gruvbox",
    },
  }
}
```
In order to, I guess, add another colorscheme and yet another file `lua/plugins/core.lua` to change the default one. Not incredibly straight forward to the layman

**Reply by dpetka2001** [reply, marked as answer]
Ok, I see where your confusion comes from. For your information everything in the docs is meant to be used as a reference. 

Also just to avoid further confusion, it doesn't really matter how you name your files. It could also be named `fdelacruz.lua`. What matters is the contents inside the file. You can have many different fragments of a plugin's spec throughout many different files. In the end they will all merge into the final plugin spec. So, for the colorscheme, what matters is that you put 
```lua
 {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "gruvbox",
    },
  }
```
inside a file and change it to whatever colorscheme you want to use. It doesn't matter if it's called `core.lua` or `colorscheme.lua`. But of course the naming of your files should make sense to you (I called mine `colorscheme.lua` cuz it makes sense to me).

I understand all this might sound like gibberish to you and that's all right. If you really want to understand how things work, you should also give [lazy.nvim](https://github.com/folke/lazy.nvim) a thorough read. It took me several thorough reads to start piecing things together.

If you have any further questions, feel free to ask. I'll try in the best of my knowledge to explain.

**Reply by t0qen** [reply]
I would like to apply catppuccin mocha theme, but how to ?

**Reply by fdelacruz** [reply]
@t0qen what have you done so far that hasn't worked?

**Reply by t0qen** [reply]
In colorscheme.lua :

return {
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin-mocha",
    },
  }
}

**Reply by fdelacruz** [reply]
This is my config:

```
return {
  {
    "catppuccin/nvim",
  },
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
    },
  },
}
```
